java/src/main/java/ai/onnxruntime/TensorInfo.java

```diff
@@ -506,8 +506,13 @@ private static void extractShape(long[] shape, int curDim, Object obj) throws Or
         throw new OrtException(
             "Supplied array is ragged, expected " + shape[curDim] + ", found " + curLength);
       }
-      for (int i = 0; i < curLength; i++) {
-        extractShape(shape, curDim + 1, Array.get(obj, i));
+      int nextDim = curDim + 1;
+      // Avoid traversing the entire array (autoboxing its values) when the next dimension is equal
+      // to the shape's length
+      if (shape.length != nextDim) {
```

Comment:
Now you've hoisted the check from line 495 up outside the for loop, you should remove the check at line 495 as it will never fail.

Reply:
done!